test/views/helpers/TextGeneratorsSpec.scala

```diff
@@ -16,42 +16,176 @@
 
 package views.helpers
 
+import forms.EmailForm
+import org.scalatest.mockito.MockitoSugar
+import org.mockito.Mockito.when
+import org.joda.time.LocalDate
 import org.scalatest.{Matchers, WordSpec}
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.i18n.{Lang, Messages}
+import uk.gov.hmrc.emailaddress.EmailAddress
 
-class TextGeneratorsSpec extends WordSpec with Matchers {
+import scala.collection.immutable
 
-  "TaxGenerators" when {
+class TextGeneratorsSpec extends WordSpec with Matchers with MockitoSugar with GuiceOneAppPerSuite {
 
-    "taxDateInterval is called" must {
-      "return the beginning and end dates of a tax year, e.g. 6 April 2018 to 5 April 2019" in {
-        TextGenerators.taxDateInterval(2018, isWelsh = false) shouldBe "6 April 2018 to 5 April 2019"
+  private val currentYear = LocalDate.now().getYear
+  private val years: immutable.Seq[Int] = (currentYear - 2 to currentYear + 3).toList
+
+
+  trait EnglishSetup {
+    implicit val englishMessages: Messages = mock[Messages]
+    when(englishMessages.lang).thenReturn(Lang("en"))
+  }
+
+  trait WelshSetup {
+    implicit val welshMessage: Messages = mock[Messages]
+    when(welshMessage.lang).thenReturn(Lang("cy"))
+  }
+
+
+
+  "TextGenerators" when {
+
+    "seperator" must {
+      "return English separator" in new EnglishSetup {
+        TextGenerator().separator shouldBe " to "
+      }
+
+      "return Welsh separator" in new WelshSetup {
+        TextGenerator().separator shouldBe " i "
+      }
+    }
+
+    "ukDateTransformer" must {
+
+      val englishList = List(
+        (new LocalDate(2016, 5, 6), "May"),
+        (new LocalDate(1993, 11, 11), "November"),
+        (new LocalDate(2020, 1, 1), "January")
+      )
+
+      englishList.foreach { dateMonthTuple =>
+        s"return month as English ${dateMonthTuple._2}" in new EnglishSetup {
+          TextGenerator().ukDateTransformer(dateMonthTuple._1) should include(dateMonthTuple._2)
+        }
+      }
+
+      val welshList = List(
```

Comment:
Since each month the logic lies in the code, you should probably have all the months in for the welsh generator.

Reply:
All months added